### PR TITLE
Support CONNECT proxies by explicitly specifying localhost instead of empty host string

### PR DESCRIPTION
**Pull request by skiptomyliu**
# TL;DR
`flyteadmin` does not support CONNECT proxies because we are not able to `no_proxy` traffic targeting localhost.   Explicitly use `localhost` in place of blank host.  This will allow us to `no_proxy` the traffic, thus traffic that's meant for localhost doesn't incorrectly egress out through the proxy.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Related upstream golang issue: https://github.com/golang/go/issues/48325

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

## Follow-up issue
_NA_

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -74,11 +74,11 @@ func SetConfig(s *ServerConfig) {
 }
 
 func (s ServerConfig) GetHostAddress() string {
-	return fmt.Sprintf(":%d", s.HTTPPort)
+	return fmt.Sprintf("localhost:%d", s.HTTPPort)
```
Comment (skiptomyliu) [inline pkg/config/config.go:77]: Dbl checking, should this listen on all interfaces, `0.0.0.0`?

**Comment by kumare3** [comment]
FlyteAdmin is not responding to healthchecks, I think local host is wrong.

**Comment by skiptomyliu** [comment]
Thanks @kumare3 , changed to listen on 0.0.0.0

**Review by kumare3 — APPROVED** [review]

**Comment by kumare3** [comment]
@skiptomyliu there are merge conflicts
